### DISCUSSION TITLE
Different preconditioners for the DavidsonSolver

### DISCUSSION THREAD

**Discussion by lelemmen** [GQCG/GQCP#432] [Ideas]
**Describe the current state of a part of the code**
Currently, the preconditioner of Davidson's diagonalization algorithm is hard-coded.

**Describe a change that would improve the code base**
It would be nice to improve the design of the `DavidsonSolver`, such that a preconditioning step can be supplied by the user.

**Additional context**
This issue is related to #332.

**Comment by lelemmen** [comment]
One of the other preconditions would be to replace any values that would yield 0 by 1, something that was mentioned in one of the ESQC lectures.

**Reply by nadia1391** [reply]
pleas can you help me to install the esqc software?
